File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.kt
class OrderedStream(n: Int) {
    private var orderedStreamArray: Array<String> = Array(n) { "0" }
    private var pointer = 0

    fun insert(idKey: Int, value: String): List<String> {
        orderedStreamArray[idKey-1] = value
        val tempList = orderedStreamArray.takeWhile { it != "0"  }.toList()
        return tempList.drop(pointer).apply { pointer = tempList.size }
    }

}

File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.py
class OrderedStream:

    def __init__(self, n: int):
        self.data = [None]*n
        self.ptr = 0 

    def insert(self, id: int, value: str) -> List[str]:
        id -= 1 
        self.data[id] = value 
        if id > self.ptr: return [] 
        
        while self.ptr < len(self.data) and self.data[self.ptr]: self.ptr += 1
        return self.data[id:self.ptr]

